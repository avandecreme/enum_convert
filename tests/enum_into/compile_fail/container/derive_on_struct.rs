use enum_convert::EnumInto;

#[derive(EnumInto)] // Should only work on enums
#[enum_into(Target)]
struct Source {
    field: i32,
}

enum Target {
    Unit,
}

fn main() {}
