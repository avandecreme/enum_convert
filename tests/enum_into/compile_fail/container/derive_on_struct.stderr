error: proc-macro derive panicked
 --> tests/enum_into/compile_fail/container/derive_on_struct.rs:3:10
  |
3 | #[derive(EnumInto)] // Should only work on enums
  |          ^^^^^^^^
  |
  = help: message: EnumInto can only be derived for enums
